packages/core/src/scanning.ts

```diff
@@ -109,3 +109,44 @@ export const scanOutputs = (
 
     return matches;
 };
+
+export const scanOutputsWithTweak = (
+    scanPrivateKey: Buffer,
+    spendPublicKey: Buffer,
+    scanTweak: Buffer,
+    outputs: Buffer[],
+    labels?: LabelMap,
+): Map<string, Buffer> => {
+    const matches = new Map<string, Buffer>();
+    let n = 0;
+    let counterIncrement = 0;
+
+    if (scanTweak.length === 33) {
+        // Use publicKeyTweakMul for compressed pubkey
+        const ecdhSecret = secp256k1.publicKeyTweakMul(
+            scanTweak,
+            scanPrivateKey,
+            true,
+        );
+        do {
+            const tweak = createTaggedHash(
+                'BIP0352/SharedSecret',
+                Buffer.concat([ecdhSecret, serialiseUint32(n)]),
+            );
+            counterIncrement = processTweak(
+                spendPublicKey,
+                tweak,
+                outputs,
+                matches,
+                labels,
+            );
+            n += counterIncrement;
+        } while (counterIncrement > 0 && outputs.length > 0);
```

Comment:
This is duplicated from `scanOutputs`, we should de-duplicate this by extracting it to a function